docs/workers.md

```diff
@@ -438,6 +438,23 @@ run_background_tasks_on: background_worker
 You might also wish to investigate the `update_user_directory` and
 `media_instance_running_background_jobs` settings.
 
+#### Notifying Application Services
+
+You can designate one worker to send output traffic to Application Services.
```

Comment:
By having a list of workers?

Reply:
Having thought about it, I'm not sure if there's much value in accepting a list right now, if we're going to assert there's only one element in the list.

The stream writers config lets you write either a string or a list of strings. We could do the same once we're ready to shard this stuff out.
The only way an advantage would arise is if we enforced it being a list now, so that we never have to deal with a string.
(Maybe this is sensible? But it doesn't seem to be that difficult to convert a `str` to `[str]` if needed; just wrap it with `_instance_to_list_converter`.)

The reason I don't think accepting a 1-long list leads to a useful/futureproof transitional config is that we will still choke on a 2-long list.
Once we're ready to accept multiple workers, we can start accepting lists at that point whilst being backwards compatible (using `_instance_to_list_converter`).

Whether someone changes their config from either
(1)
```yaml
notify_appservices_from_worker: worker1
```
or (2)
```yaml
notify_appservices_from_worker:
  - worker1
```
to
```yaml
notify_appservices_from_worker:
  - worker1
  - worker2
```
, it's going to be backwards-incompatible since the older Synapse version won't understand how to shard the task.

I may well be missing something; (and perhaps we DO want things to be a list by default) -— let me know if you have any other thoughts

Reply:
I don't feel strongly about it either way. It was more of an idle thought. If you don't see a benefit, I would say to keep things simple for now!

Reply:
Fair enough! It was definitely worth consciously thinking about though, so good call anyway